### PR TITLE
feat: support Chunk.groupsIterable and ChunkGroup.getParents

### PR DESCRIPTION
**Pull request by bvanjoi**
This PR aims to address issue #3896 by exclusively using ﻿`u32` in `﻿JsChunkGroup::parents` and `﻿JsChunk::groups`, thereby reducing overhead.

### CODE REVIEW

**Comment by 9aoy** [comment]
!canary

**Comment by 9aoy** [comment]
!canary

**Comment by 9aoy** [comment]
!canary

**Comment by 9aoy** [comment]
need chunkGroup.name
image


img_v3_0258_9cbee020-af08-44c0-b6be-2134b35d0e8g

File: crates/rspack/tests/tree-shaking/webpack-inner-graph-export-default-named/snapshot/new_treeshaking.snap [hunk 1]
```diff
@@ -158,7 +158,7 @@ __webpack_require__.r(__webpack_exports__);
 
 it("should generate valid code", ()=>{
     expect((0, _a__WEBPACK_IMPORTED_MODULE_0__["default"])()).toBe("x");
-    expect(new (0, _d__WEBPACK_IMPORTED_MODULE_3__["default"])().method()).toBe("x");
+    expect(new _d__WEBPACK_IMPORTED_MODULE_3__["default"]().method()).toBe("x");
```
Comment (bvanjoi) [inline crates/rspack/tests/tree-shaking/webpack-inner-graph-export-default-named/snapshot/new_treeshaking.snap:161]: I think that the PR wouldn't affect the output, yet it's actually happening...

However, it seems to be acceptable.

File: crates/rspack/tests/tree-shaking/webpack-inner-graph-export-default-named/snapshot/output.snap [hunk 2]
```diff
@@ -158,7 +158,7 @@ __webpack_require__.r(__webpack_exports__);
 
 it("should generate valid code", ()=>{
     expect((0, _a__WEBPACK_IMPORTED_MODULE_0__["default"])()).toBe("x");
-    expect(new (0, _d__WEBPACK_IMPORTED_MODULE_3__["default"])().method()).toBe("x");
+    expect(new _d__WEBPACK_IMPORTED_MODULE_3__["default"]().method()).toBe("x");
```
Comment (bvanjoi) [inline crates/rspack/tests/tree-shaking/webpack-inner-graph-export-default-named/snapshot/output.snap:161]: same as above

**Review by 9aoy — APPROVED** [review]

**Pull request merged**